### PR TITLE
server,autoconfig: automatic configuration via config tasks

### PR DESCRIPTION
**Pull request by knz**
Epic: CRDB-23559
Informs #98431.
All commits but the last are from #98993.

This change introduces "auto config tasks", a mechanism through which
configuration payloads ("tasks") can be injected into a running SQL
service.

This is driven via the "auto config runner" job that was introduced in
the previous commit. The job listens for the arrival of new task
definitions via a `Provider` interface. When new tasks are known, and
previous tasks have completed, the runner creates a job for the first
next task.

Release note: None

### CODE REVIEW

**Comment by knz** [comment]
@ajwerner I plan to add the additional layer for "pingback" of results in a later PR as discussed in #98431.

File: pkg/server/autoconfig/auto_config.go
Comment (ajwerner) [inline pkg/server/autoconfig/auto_config.go:164]: It sort of feels like it'd be nice to combine the fetching of the currently started task with the fetching of the latest completed task.

File: pkg/server/autoconfig/doc.go
Comment (ajwerner) [inline pkg/server/autoconfig/doc.go:94]: It seems to me that job success is at most once. If these tasks fail with a permanent error, they are not retried by anything. Cancellation is a mechanism to cause failure. One thing we might want to expose is a way to mark a task as required. In jobs parlance, this would mean that it's non-cancelable. A non-cancelable job treats errors as retryable by default (as opposed to permanent), and it prevents user cancelation. I suspect for the hard-coded tasks, we'll want to mark them as uncancelable.

File: pkg/server/autoconfig/task_markers.go
Comment (ajwerner) [inline pkg/server/autoconfig/task_markers.go:57]: why are you passing an `execCfg` and then not using it?

I view an execCfg argument to a function as a code smell

File: pkg/server/autoconfig/task_markers.go
Comment (ajwerner) [inline pkg/server/autoconfig/task_markers.go:76]: why are you passing an `execCfg` and then not using it?

File: pkg/server/autoconfig/task_markers.go
Comment (ajwerner) [inline pkg/server/autoconfig/task_markers.go:118]: why are you passing an `execCfg` and then not using it?

File: pkg/server/autoconfig/task_markers.go
Comment (ajwerner) [inline pkg/server/autoconfig/task_markers.go:152]: why are you passing an `execCfg` and then not using it?

File: pkg/server/autoconfig/doc.go
Comment (ajwerner) [inline pkg/server/autoconfig/doc.go:70]: One thing I'm not seeing in this PR is how we talk about different environments and the completion of their tasks independently. I expected to see some sort of environment ID or long-running job per environment or something like that.

Is it the responsibility of the enviroments to make sure that the IDs are monotonically increasing just within that environment, or across all environments?

File: pkg/server/autoconfig/acprovider/provider.go
Comment (ajwerner) [inline pkg/server/autoconfig/acprovider/provider.go:36]: This doesn't implement the contract `The channel receives an initial event immediately.`

Consider revisiting that.

File: pkg/server/autoconfig/acprovider/provider.go
Comment (ajwerner) [inline pkg/server/autoconfig/acprovider/provider.go:17]: Something about this interface feels a little off. I guess it's the fact that it always hands you all the tasks, and that it seems to be something you tell about task completion, but it doesn't have a clear contract around how that affects behavior. The channel-based API also feels a bit fishy. 

Is it important for the client to get more than one task at a time?
What about something like:

```
// Peek will block waiting for the first task the provider believes still needs to be run.
// It will return an error if the context is canceled while waiting.
Peek(ctx context.Context) (autoconfigpb.Task, error)

// Pop will report the completion of all tasks with ID up to completed.
Pop(ctx context.Context, completed TaskID) error
```

**Review by ajwerner — COMMENTED** [review]
The design of the subsystem seems reasonable. I feel like the code could use some polish in terms of dependency definition and plumbing, but it's understandable enough. I'd consider making a structure that can be dependency injected to interact with the job data and what not and then take that also as an interface into the job and task.

File: pkg/server/autoconfig/task_markers.go
Comment (adityamaru) [inline pkg/server/autoconfig/task_markers.go:63]: @knz maybe I'm missing something but can we not initialize an `InfoStorage` after loading the job with `jobs.AutoConfigRunnerJobID`. So something like:

```
job, err := jobRegistry.LoadJobWithTxn(ctx, jobs.AutoConfigRunnerJobID, txn)
if err != nil {
    return err
}
infoStorage := job.InfoStorage(txn)
infoStorage.Write(startMarkerInfoKey, []byte(strconv.FormatUint(uint64(jobID), 10)))
```

Our `Write` method on `InfoStorage` under the hood, is a delete of the existing row corresponding to the `job_id, info_key` and an insert of the new row. The reasons for this are explained here - https://github.com/cockroachdb/cockroach/pull/82638/files#diff-d65357c15cffdb009c1b3fd0f6456265d25b94758f326207c531b8f734b59164R328-R348. Different parts of the job system are going to rely on their being exactly one row corresponding to a `<job_id, info_key>` so I'd like to discourage direct SQL manipulation of the `job_info` table.

File: pkg/server/autoconfig/task_markers.go
Comment (adityamaru) [inline pkg/server/autoconfig/task_markers.go:82]: Can we use the `Iterate` method on InfoStorage to go over all the rows corresponding to `jobs.AutoConfigRunnerJobID` and with an info_key prefix == InfoKeyStartPrefix?

Comment (knz) [reply pkg/server/autoconfig/task_markers.go:82]: huh no. There could be thousands of them. We are using the index here to get directly to the latest one.

File: pkg/server/autoconfig/task_markers.go
Comment (knz) [reply pkg/server/autoconfig/task_markers.go:63]: > can we not initialize an InfoStorage after loading the job

I would like to avoid re-loading the permanent job over and over in every task. It's also silly given that the job entry (and its payload, and its progress) are not needed here, just its ID to populate the job_info table.

File: pkg/server/autoconfig/task_markers.go
Comment (knz) [reply pkg/server/autoconfig/task_markers.go:162]: ok, that i will do.

File: pkg/server/autoconfig/task_markers.go
Comment (adityamaru) [reply pkg/server/autoconfig/task_markers.go:63]: makes sense, I just realized that you might not need to load it. You can just:

```
j := &jobs.Job{id: jobs.AutoConfigRunnerJobID}
infoStorage := j.InfoStorage(txn)
...
```

File: pkg/server/autoconfig/task_markers.go
Comment (adityamaru) [reply pkg/server/autoconfig/task_markers.go:82]: Makes sense, thanks

**Review by knz — COMMENTED** [review]
>  I feel like the code could use some polish in terms of dependency definition and plumbing, but it's understandable enough. I'd consider making a structure that can be dependency injected to interact with the job data and what not and then take that also as an interface into the job and task.

I will do this in the next iteration.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-:-NQvy76a0Puie8u6sR44:bm40itr)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @ajwerner)

___
*[`pkg/server/autoconfig/auto_config.go` line 164 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Js-r2-164:-NQunBxJ-IfURw57PSz8:bpnhyet) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/auto_config.go#L164)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

It sort of feels like it'd be nice to combine the fetching of the currently started task with the fetching of the latest completed task. 
</blockquote></details>

I don't see it - one of them it used in two places, the other just one; also they access different segments of the keyspace so will result in two separate storage scans in any case.
___
*[`pkg/server/autoconfig/doc.go` line 70 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q6-r2-70:-NQunKNiDfSQqD0WhRG_:b-ycpcil) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/doc.go#L70)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

One thing I'm not seeing in this PR is how we talk about different environments and the completion of their tasks independently. I expected to see some sort of environment ID or long-running job per environment or something like that.

Is it the responsibility of the enviroments to make sure that the IDs are monotonically increasing just within that environment, or across all environments?
</blockquote></details>

Done - clarified that each environment has a separate sequence.
___
*[`pkg/server/autoconfig/doc.go` line 94 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q6-r2-94:-NQvxnyo8aFprsAZgqYC:b-7zcqlm) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/doc.go#L94)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

It seems to me that job success is at most once. If these tasks fail with a permanent error, they are not retried by anything. Cancellation is a mechanism to cause failure. One thing we might want to expose is a way to mark a task as required. In jobs parlance, this would mean that it's non-cancelable. A non-cancelable job treats errors as retryable by default (as opposed to permanent), and it prevents user cancelation. I suspect for the hard-coded tasks, we'll want to mark them as uncancelable. 
</blockquote></details>

I see where this is coming from but I'd like to delay the introduction of this until we get some experience with the mechanism. In a first release, the hard-coded task (config profiles) can accommodate tasks that don't get executed due to errors. We can inspect their status by other means.
___
*[`pkg/server/autoconfig/task_markers.go` line 57 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-57:-NQudUDO1FjedqK8Hxsk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L57)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

why are you passing an `execCfg` and then not using it?

I view an execCfg argument to a function as a code smell
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/task_markers.go` line 63 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-63:-NQvuFf17R293mBsmt47:b-1uvhw0) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L63)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

makes sense, I just realized that you might not need to load it. You can probably:

```
j := &jobs.Job{id: jobs.AutoConfigRunnerJobID}
infoStorage := j.InfoStorage(txn)
...
```

edit: noticed the id field is not exported so you'd have to add a method to create a Job struct if you want to follow this approach
</blockquote></details>

ok done #98993
___
*[`pkg/server/autoconfig/task_markers.go` line 76 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-76:-NQudXTrBqrmFVlRlrM3:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L76)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

why are you passing an `execCfg` and then not using it?
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/task_markers.go` line 82 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-82:-NQvuU1u-YdCZfzPHWnp:b-iz0o8s) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L82)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

Makes sense, thanks
</blockquote></details>

see what i did in #98993.
___
*[`pkg/server/autoconfig/task_markers.go` line 118 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-118:-NQud_XhEwrvc1FbvboQ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L118)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

why are you passing an `execCfg` and then not using it?
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/task_markers.go` line 152 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-152:-NQudaDU6ZewLCyDVxQh:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L152)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

why are you passing an `execCfg` and then not using it?
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/task_markers.go` line 162 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r2-162:-NQvuWEY5fuel6-ml56H:bbqr1xy) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/task_markers.go#L162)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

ok, that i will do.
</blockquote></details>

#98993
___
*[`pkg/server/autoconfig/acprovider/provider.go` line 17 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Jx-r2-17:-NQudcZ75c5ZyZUOMzGz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/acprovider/provider.go#L17)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

Something about this interface feels a little off. I guess it's the fact that it always hands you all the tasks, and that it seems to be something you tell about task completion, but it doesn't have a clear contract around how that affects behavior. The channel-based API also feels a bit fishy. 

Is it important for the client to get more than one task at a time?
What about something like:

```
// Peek will block waiting for the first task the provider believes still needs to be run.
// It will return an error if the context is canceled while waiting.
Peek(ctx context.Context) (autoconfigpb.Task, error)

// Pop will report the completion of all tasks with ID up to completed.
Pop(ctx context.Context, completed TaskID) error
```
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/acprovider/provider.go` line 36 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Jx-r2-36:-NQuddgK4_bQAnJgf1Cz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/acprovider/provider.go#L36)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

This doesn't implement the contract `The channel receives an initial event immediately.`

Consider revisiting that.
</blockquote></details>

Done.

**Review by adityamaru — COMMENTED** [review]
I did a first pass, this is really great stuff. I only have questions so everything LGTM once those are cleared up.

**Review by knz — COMMENTED** [review]
> This file seems unit testable, we can do it as a follow-up but I think some tests to exercise the encoding/decoding would be nice.

Done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-:-NRUrXg96eYPbNDhY_1V:b8b1rqz)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @ajwerner)

___
*[`pkg/server/autoconfig/auto_config.go` line 66 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Js-r4-66:-NRUUYHQ69qMUuVsR4NP:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config.go#L66)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

q: is this a temporary work around while we figure out the registry-level fix? If it is, should we add a TODO?
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/auto_config_env_runner.go` line 236 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQud89erWVHddZZiK5j-r4-236:-NRUP5iu0F2Tks1jXTq9:bc41cfy) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config_env_runner.go#L236)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

The comment on `Run` states:
```
// The assumption is that these jobs were created by
// this registry and thus are pre-claimed by it.
```

I don't think that there is any guarantee that `prevJobID` was created by this node's registry. If the assumption doesn't hold, my read of the code is that the waiting set won't get notified on job completion since it isn't being run by the current registry?
</blockquote></details>

I had the same concern originally ([internal convo](https://cockroachlabs.slack.com/archives/C02DSDS9TM1/p1678478253603559?thread_ts=1678477481.702839&cid=C02DSDS9TM1))

Bottom line is `Run` falls back to polling if the job is not locally registered already.
___
*[`pkg/server/autoconfig/doc.go` line 94 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q6-r2-94:-NRUTpiN3OykSUPZHaHs:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2535e7e1162155554bad9ad796f2a01aa9adff39/pkg/server/autoconfig/doc.go#L94)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

+1, maybe we add a TODO for now @knz to capture this idea of non-cancellable, required tasks?
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/task_markers.go` line 198 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQMFgY5g6x6seQhf9Q7-r4-198:-NRUM8YPDF1NgWvpGCKk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/task_markers.go#L198)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

stale comment?
</blockquote></details>

Done.
___
*[`pkg/server/autoconfig/acprovider/provider.go` line 42 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Jx-r4-42:-NRUMBPsESOWgty3f21o:bcw4564) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/acprovider/provider.go#L42)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

I currently see only one caller of Pop when the auto config runner sees the last completed task > the task we are processing. Do we expect each task runner to Pop the task on completion?
</blockquote></details>

It's not strictly necessary (the env runner will pick up the completion marker at the next round), but I guess it doesn't hurt. Done.
___
*[`pkg/server/autoconfig/auto_config_test.go` line 141 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Jy-r4-141:-NRUP-5qCS00Ik-_SKAa:bbgeriz) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config_test.go#L141)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

nit: maybe a quick comment explaining why it is disabled.
</blockquote></details>

actually there was no good reason. Removed.

**Comment by knz** [comment]
@adityamaru I've rebased this on top of the other changes to change the info key to string. RFAL

**Comment by adityamaru** [comment]
Reviewing this is on my list of things for tomorrow. Sorry for the delay @knz

**Review by adityamaru — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98459)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @knz)

___
*[`pkg/server/autoconfig/auto_config.go` line 66 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Js-r4-66:-NS69jJoARTJ66JCBMHJ:bmostn1) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config.go#L66)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Given that 99242 has been addressed can we remove this?



_Code quote:_
```Go
https://github.com/cockroachdb/cockroach/issues/99242
```
___
*[`pkg/server/autoconfig/auto_config_env_runner.go` line 236 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQud89erWVHddZZiK5j-r4-236:-NS69p4R7Pr8VbRTc4Ej:b-jpvrvo) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config_env_runner.go#L236)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

I had the same concern originally ([internal convo](https://cockroachlabs.slack.com/archives/C02DSDS9TM1/p1678478253603559?thread_ts=1678477481.702839&cid=C02DSDS9TM1))

Bottom line is `Run` falls back to polling if the job is not locally registered already.
</blockquote></details>

perfect, thanks!
___
*[`pkg/server/autoconfig/auto_config_test.go` line 141 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Jy-r4-141:-NS6AHBA6Rz3QaMGE6hv:bt3rnve) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config_test.go#L141)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

actually there was no good reason. Removed.
</blockquote></details>

nit: leftover commented code

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98459)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @ajwerner)

___
*[`pkg/server/autoconfig/auto_config.go` line 66 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Js-r4-66:-NS7VxMcDugdM7VtEJdH:b3l801l) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config.go#L66)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

Given that 99242 has been addressed can we remove this?
</blockquote></details>

Done. good call!
___
*[`pkg/server/autoconfig/auto_config_test.go` line 141 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/98459#-NQLho6esJk4zGier2Jy-r4-141:-NS7W-R0AATWCxTno73O:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/b36f0f2c23a5257f6c6f612e938e4da97f4e9a7e/pkg/server/autoconfig/auto_config_test.go#L141)):*
<details><summary><i>Previously, adityamaru (Aditya Maru) wrote…</i></summary><blockquote>

nit: leftover commented code
</blockquote></details>

Fixed.

**Comment by knz** [comment]
bors r=adityamaru

**Pull request merged**
